src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+        """
+
+        if self.__is_url(message):
+            # The message is a url
+            if re.search(r'(playlist)', message):
+                # The message is a playlist
+                return 0
+            else:
+                return 1
+        else:
+            # The message is a string
+            return 2
+
+    def __get_opus_stream(self, formats):
+        """
+        Get the opus formatted streaming link from the formats dictionary.
+        :param formats: The formats dictionary that contains the different streaming links.
+        :return: A streaming url that links to an opus stream and the bit rate of the stream.
+        """
+
+        # Limit the codecs to just opus, as that is required for streaming audio
+        opus_formats = [x for x in formats if x.get('acodec') == 'opus']
+
+        # Sort the formats from highest br to lowest
+        sorted_opus = list(sorted(opus_formats, key=lambda k: float(k.get('abr')), reverse=True))
+
+        chosen_stream = sorted_opus[self._bitrate_quality]
+
+        return chosen_stream.get('url'), chosen_stream.get('abr')
+
+    def __format_download_data(self, download_data):
+        """
+        Format a songs data to remove the useless data.
+        :param download_data: The song data to format.
+        :return: A dictionary of data which is a subset of the param download_data
+        """
+
+        stream, rate = self.__get_opus_stream(download_data.get('formats'))
+        useful_data = {'title': download_data.get('title'), 'id': download_data.get('id'),
+                       'link': download_data.get('webpage_url'), 'length': download_data.get('duration'),
+                       'stream': stream,
+                       'bitrate': rate,
+                       'thumbnail': self.__get_thumbnail(download_data.get('thumbnails')),
+                       'filename': download_data.get('filename')}
+        return useful_data
+
+    def __get_thumbnail(self, thumbnails):
+        """
+        Get the thumbnail from url from the thumbnails dictionary.
+        :param thumbnails: The dictionary of thumbnail urls.
+        :return: A url linking to the video thumbnail.
+        """
+
+        # Sort by thumbnail size
+        sorted_thumbnails = list(sorted(thumbnails, key=lambda k: int(k.get('height')), reverse=True))
+
+        return sorted_thumbnails[0].get('url')
+
+    def __is_url(self, string):
+        """
+        Returns if the string given is a url.
+        :param string: The string to check.
+        :return: True if the string is a url. False otherwise.
+        """
+
+        # Match desktop, mobile and playlist links
+        re_desktop = r'(http[s]?://)?youtube.com/(watch\?v)|(playlist\?list)='
+        re_mobile = r'(http[s]?://)?youtu.be/([a-zA-Z]|[0-9])+'
+
+        re_string = "(" + re_desktop + ") | (" + re_mobile + ")"
+
+        if re.search(re_desktop, string) or re.search(re_mobile, string):
```

Comment:
you could `return bool(re.search(re_desktop, string) or re.search(re_mobile, string))` but that's neither here nor there